core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -35,6 +37,8 @@
  * An abstraction over files and classpath resources aimed at encapsulating all the complexity of generating
  * a path that the Docker daemon is about to create a volume mount for.
  */
+@EqualsAndHashCode(of = {"path", "forcedFileMode"})
```

Comment:
```suggestion
@EqualsAndHashCode(of = "path")
```

Reply:
Yes, but wouldn't that be confusing. They _are_ different, after all, if their file mode is different. Otherwise, only the path should be the key.

Reply:
> They are different, after all, if their file mode is different

they represent the same path. The forced file mode is only a hint that is used to override the one from the file system (but the target object is the same)

Reply:
I don't agree, but it's more your code than mine ;-)

Reply:
I just pushed another update